web/app/js/components/NetworkGraph.jsx

```diff
@@ -0,0 +1,217 @@
+
+    this.loadFromServer();
+  }
+
+  componentDidUpdate() {
+    simulation.alpha(1).restart();
+  }
+
+  componentWillUnmount() {
+    this.api.cancelCurrentRequests();
+  }
+
+  loadFromServer() {
+    if (this.state.pendingRequests) {
+      return; // don't make more requests if the ones we sent haven't completed
+    }
+    this.setState({ pendingRequests: true });
+
+    let deployments = _.sortBy(_.map(this.props.deployments, 'name'));
+    let urls = _.map(this.props.deployments, d => {
+      return this.api.fetchMetrics(this.api.urlsForResource("deployment", "") + "&to_name=" + d.name);
+    });
+
+    this.api.setCurrentRequests(urls);
+
+    Promise.all(this.api.getCurrentPromises())
+      .then(results => {
+        let links = [];
+        let nodeList = [];
+
+        _.map(results, (r, i) => {
+          let rows = _.get(r, ["ok", "statTables", 0, "podGroup", "rows"]);
+          let dst = deployments[i];
+          _.map(rows, row => {
+            links.push({
+              source: row.resource.name,
+              target: dst,
+              value: 1
+            });
+            nodeList.push(row.resource.name);
+            nodeList.push(dst);
+          });
+        });
+
+        let nodes = _.map(_.uniq(nodeList), n => {
+          return { id: n, r: 15 };
+        });
+
+        this.setState({
+          nodes: nodes,
+          links: links,
+          pendingRequests: false,
+          error: ""
+        });
+      })
+      .then( () => this.updateGraph())
+      .catch(this.handleApiError);
+  }
+
+  updateGraph() {
+    this.svg.append("svg:defs").selectAll("marker")
+      .data(["end"])      // Different link/path types can be defined here
+      .enter().append("svg:marker")    // This section adds in the arrows
+      .attr("id", String)
+      .attr("viewBox", "0 -5 10 10")
+      .attr("refX", 18)
+      .attr("refY", -0.25)
+      .attr("markerWidth", 3)
+      .attr("markerHeight", 3)
+      .attr("orient", "auto")
+      .append("svg:path")
+      .attr("d", "M0,-5L10,0L0,5");
+
+    // add the links and the arrows
+    const path = this.svg.append("svg:g").selectAll("path")
+      .data(this.state.links)
+      .enter().append("svg:path")
+      .attr("stroke-width", 3)
+      .attr("stroke", "#000000")
+      .attr("marker-end", "url(#end)");
+
+    const nodeElements = this.svg.append('g')
+      .selectAll('circle')
+      .data(this.state.nodes)
+      .enter().append('circle')
+      .attr('r', 10)
+      .attr('fill', 'steelblue')
+      .call(d3.drag()
+        .on("start", this.dragstarted)
+        .on("drag", this.dragged)
+        .on("end", this.dragended));
+
+    const textElements = this.svg.append('g')
+      .selectAll('text')
+      .data(this.state.nodes)
+      .enter().append('text')
+      .text(node => node.id)
+      .attr('font-size', 15)
+      .attr('dx', 15)
+      .attr('dy', 4);
+
+    simulation.nodes(this.state.nodes).on("tick", () => {
```

Comment:
I played around with making the nodes a bit bigger; it did involve some tweaking to get the arrows not to be buried. wdyt?

```
@@ -140,7 +140,10 @@ class NetworkGraph extends React.Component {
       .selectAll('circle')
       .data(this.state.nodes)
       .enter().append('circle')
-      .attr('r', 10)
+      .attr('r', d => {
+        d.radius = 15;
+        return d.radius;
+      })
       .attr('fill', 'steelblue')
       .call(d3.drag()
         .on("start", this.dragstarted)
@@ -157,12 +160,21 @@ class NetworkGraph extends React.Component {
       .attr('dy', 4);

     simulation.nodes(this.state.nodes).on("tick", () => {
+
+
       path
-        .attr("d", node =>  "M" +
+        .attr("d", node =>  {
+          let diffX = node.target.x - node.source.x;
+          let diffY = node.target.y - node.source.y;
+          let pathLength = Math.sqrt(diffX*diffX + diffY*diffY);
+          let offsetX = (diffX * node.target.radius) / pathLength;
+          let offsetY = (diffY * node.target.radius) / pathLength;
+          return "M" +
               node.source.x + " " +
               node.source.y + " L " +
-              node.target.x + " " +
-              node.target.y);
+              (node.target.x - offsetX) + " " +
+              (node.target.y- offsetY);
+        });
```